doc/tutorials/imgproc/histograms/template_matching/template_matching.markdown

```diff
@@ -1,28 +1,75 @@
 Template Matching {#tutorial_template_matching}
 =================
 
+@prev_tutorial{tutorial_back_projection}
+@next_tutorial{tutorial_find_contours}
+
 Goal
 ----
 
 In this tutorial you will learn how to:
 
+@add_toggle_cpp
+
 -   Use the OpenCV function @ref cv::matchTemplate to search for matches between an image patch and
     an input image
 -   Use the OpenCV function @ref cv::minMaxLoc to find the maximum and minimum values (as well as
     their positions) in a given array.
 
+@end_toggle
+
+@add_toggle_java
```

Comment:
Is there any way to avoid text duplication? This is the same paragraph copied three times!
I suggest adding inline commands (`@toggle_cpp cv::matchTemplate @toggle_java Imgproc.MatchTemplate() @toggle_python cv2.matchTemplate() @end_toggle`) to avoid text duplication. Or maybe you have a better idea?